agent/internal/client/request.go

```diff
@@ -156,7 +157,28 @@ func (c *communicatorImpl) retryRequest(ctx context.Context, info requestInfo, d
 
 	r.Header.Add(evergreen.ContentLengthHeader, strconv.Itoa(len(out)))
 
-	return utility.RetryRequest(ctx, r, c.maxAttempts, c.timeoutStart, c.timeoutMax)
+	var resp *http.Response
+	resp, err = utility.RetryRequest(ctx, r, c.maxAttempts, c.timeoutStart, c.timeoutMax)
+	if err != nil && resp != nil && resp.StatusCode == 400 {
+		var taskId, start, end string
+		if info.taskData != nil {
+			taskId = info.taskData.ID
+		}
+		if len(out) >= 100 {
+			start = string(out[0:100])
```

Comment:
@ablack12  nit: These log only if the length is greater than 100. They are empty if the length is less than 100. I'm not sure if there are messages shorter than 100 characters, but if there are, I think this wouldn't log at all.